packages/design-picker/src/hooks/use-generated-designs-query.ts

```diff
@@ -0,0 +1,35 @@
+import { useQuery, UseQueryResult } from 'react-query';
+import wpcom from 'calypso/lib/wp'; // eslint-disable-line no-restricted-imports
+import type { BlockRecipe, Design } from '../types';
+
+export function useGeneratedDesignsQuery(): UseQueryResult< Design[] > {
+	return useQuery( [ 'generated-designs' ], () => fetchBlockRecipes(), {
+		select: ( response ) => response.map( apiBlockRecipeToDesign ),
+		enabled: true,
+		staleTime: Infinity,
+	} );
+}
+
+function fetchBlockRecipes(): Promise< BlockRecipe[] > {
+	return wpcom.req.get( {
+		apiNamespace: 'wpcom/v2',
+		path: '/block-recipes',
+	} );
+}
+
+function apiBlockRecipeToDesign( recipe: BlockRecipe ): Design {
+	const { slug, title, stylesheet, pattern_ids } = recipe;
+
+	return {
+		slug,
+		title,
+		recipe: { stylesheet, patternIds: pattern_ids },
+		categories: [ { slug, name: title } ],
```

Comment:
Do we still need generated designs to have a category? Where is it being used?

Reply:
Not used but same as `features`, `template`, and `theme` it's currently [not an optional property](https://github.com/Automattic/wp-calypso/blob/trunk/packages/design-picker/src/types.ts#L29).

Reply:
I see, I'd suggest set is as `[]` then to reduce confusion?

Reply:
`slug` and `name` are also non-optional 🥲

Reply:
I mean

```
categories: [],
```

Reply:
Ah, okay 👍